Source/HTTP/WinHttp/winhttp_http_task.cpp

```diff
@@ -785,11 +962,79 @@ http_internal_wstring flatten_http_headers(_In_ const http_header_map& headers)
     return flattened_headers;
 }
 
+HRESULT winhttp_http_task::query_security_information(
+    http_internal_wstring wUrlHost
+    )
+{
+#if HC_PLATFORM == HC_PLATFORM_GDK
+
+    XAsyncBlock asyncBlock{};
+    asyncBlock.queue = m_env->GetImmediateQueue();
+    HRESULT hr = XNetworkingQuerySecurityInformationForUrlUtf16Async(wUrlHost.c_str(), &asyncBlock);
+    if (FAILED(hr))
+    {
+        HC_TRACE_ERROR(HTTPCLIENT, "winhttp_http_task [ID %llu] [TID %ul] XNetworkingQuerySecurityInformationForUrlUtf16Async 0x%0.8x", TO_ULL(HCHttpCallGetId(m_call)), GetCurrentThreadId(), hr);
+        return hr;
+    }
+
+    hr = XAsyncGetStatus(&asyncBlock, true);
```

Comment:
>hr = XAsyncGetStatus(&asyncBlock, true); [](start = 4, length = 40)

fine to block here?

Reply:
Yeah, Doug recommended this approach with immediate queue as he does this approach elsewhere